src/mlpack/core/data/pre_processing.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file core/data/pre_processing_impl.hpp
+ * @author Gopi Manohar Tatiraju
+ *
+ * Generic dataframe-like class for C++ for data pre processing
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DATA_PRE_PROCESSING_HPP
+#define MLPACK_CORE_DATA_DATA_PRE_PROCESSING_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A datafram-like class for pre-processing the dataset 
+ * before feeding it to any model. 
+ *  
+ * @param path Path to the csv file
+ * @param headers Headers of the dataset
+ * @param numericAttributes Bool array to indicate numeric attributes
+ * @param numericData Arma mat of numeric attributes only
+ */
+class Data {
```

Comment:
So my primary suggestion here is that this class is basically storing data and the set of dimensions that are categorical.  But we already have basically this in the `data::DatasetInfo` class.  How can we redo this code so that it fits (or adapts) that existing support that we already have?  I don't want to end up with two very different ways of achieving the same support---that makes the library feel schizophrenic and not unified. :+1:

Reply:
Heyy, first of all, sorry for the late reply and not following up, okay, I will try and redo the code so that it can fit here...